session.go

```diff
@@ -73,6 +69,8 @@ type session struct {
 	// closeChan is used to notify the run loop that it should terminate.
 	// If the value is not nil, the error is sent as a CONNECTION_CLOSE.
 	closeChan chan *qerr.QuicError
+	// the error this session was closed with
+	closeErr  error
```

Comment:
I know, but I don't understand why that means we require a second way of delivering errors.